internal/snmp/translate.go

```diff
@@ -0,0 +1,183 @@
+package snmp
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/sleepinggenius2/gosmi"
+	"github.com/sleepinggenius2/gosmi/types"
+)
+
+// must init, append path for each directory, load module for every file
+// or gosmi will fail without saying why
+func LoadMibsFromPath(paths []string, log telegraf.Logger) error {
+	gosmi.Init()
```

Comment:
I have also added a mutex

Reply:
Hey @MyaLongmire, I don't see any check... When calling `gosmi.Init()` it calls [this function](https://github.com/sleepinggenius2/gosmi/blob/841095eac905c34222076a761d6d534d1fc8fef6/config.go#L9) which in turn calls `smi.Init()` that can be found [here](https://github.com/sleepinggenius2/gosmi/blob/841095eac905c34222076a761d6d534d1fc8fef6/smi/config.go#L41. Already at this point, multiple files are reread etc. Additionally `internal.Init()` is called also modifying data. While I agree that currently nothing "critical" is changed there, it won't hurt to be on the safe side by
```suggestion
	once.Do(gosmi.Init())
```
and place `var once sync.Once` in the package scope, to be sure we call that function only once at least in that package. It would be even better to make sure it's not called elsewhere, but that's probably out of scope for this PR...

Reply:
This is added in[ pr 10199](https://github.com/influxdata/telegraf/pull/10199)

Reply:
gosmi.Init() always calls smi.Init("gosmi") which always calls internal.Init(handleName) with the same handleName. That function does check whether it already has an internal handle of that name. That's why it looked to me like calling gosmi.Init() multiple times is safe.

It's a good idea to be defensive in our code and add sync.Once() anyway. I'm not sure gosmi intends it to be safe to call gosmi.Init() multiple times. I didn't see a guarantee in the docs, so even if it is now it might not be in a future version of gosmi.